api/src/main/java/org/openmrs/module/cohort/CohortMemberAttributeType.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.cohort;
+
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.ToString;
+import org.openmrs.BaseOpenmrsData;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.Table;
+
+@Entity
+@Table(name = "cohort_member_attribute_type")
+@EqualsAndHashCode(callSuper = true)
+@Data
+@ToString
+public class CohortMemberAttributeType extends BaseOpenmrsData {
```

Comment:
Does it make sense to use the built-in `BaseAttributeType`, `BaseAttribute`, and `BaseCustomizableData` classes for `CohortMemberAttributeType`, `CohortMemberAttribute` and `CohortMember` respectively? This provides `CohortMember` with a link to it's own `CohortMemberAttribute`s and while the model is a little more complex, most of that complexity can simply be ignored.